clippy_lints/src/manual_float_methods.rs

```diff
@@ -0,0 +1,175 @@
+use clippy_utils::{
+    consts::{constant, Constant},
+    diagnostics::span_lint_and_then,
+    is_from_proc_macro, path_to_local,
+    source::snippet_opt,
+};
+use rustc_errors::Applicability;
+use rustc_hir::{BinOpKind, Expr, ExprKind};
+use rustc_lint::{LateContext, LateLintPass, Lint, LintContext};
+use rustc_middle::lint::in_external_macro;
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for manual `is_infinite` reimplementations
+    /// (i.e., `x == <float>::INFINITY || x == <float>::NEG_INFINITY`).
+    ///
+    /// ### Why is this bad?
+    /// This should use the aforementioned dedicated method instead, `is_infinite`.
```

Comment:
Each paragraph usually stands alone, usually I wouldn't mention it, but this sentence sounds weird to me anyways.

```suggestion
    /// The method `is_infinite` is shorter and more readable.
```